src/frontend/web_application/src/modules/notification/components/MessageNotificationHandler/presenter.jsx

```diff
@@ -0,0 +1,77 @@
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withI18n } from '@lingui/react';
+import isequal from 'lodash.isequal';
+import { withSettings } from '../../../../modules/settings';
+import { notify as browserNotify } from '../../../../services/browser-notification';
+
+@withI18n()
+@withSettings()
+class MessageNotificationHandler extends Component {
+  static propTypes = {
+    requestDiscussions: PropTypes.func.isRequired,
+    requestNewMessages: PropTypes.func.isRequired,
+    newMessageIds: PropTypes.arrayOf(PropTypes.string),
+    settings: PropTypes.shape({}),
+    i18n: PropTypes.shape({ _: PropTypes.func.isRequired }).isRequired,
+  };
+
+  static defaultProps = {
+    newMessageIds: [],
+    settings: undefined,
+  };
+
+  state = {
+    hasActivity: false,
+  };
+
+  componentDidMount() {
+    this.handleNotifications(this.props);
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
`componentWillReceiveProps` is deprecated and will be dropped after React 17, see : https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops